src/tools/kcSanitize/KcSanitizer.ts

```diff
@@ -0,0 +1,67 @@
+import { KcSanitizerPolicy } from "keycloakify/tools/kcSanitize/KcSanitizerPolicy";
+
+// implementation of keycloak java sanitize method ( KeycloakSanitizerMethod )
+// https://github.com/keycloak/keycloak/blob/8ce8a4ba089eef25a0e01f58e09890399477b9ef/services/src/main/java/org/keycloak/theme/KeycloakSanitizerMethod.java#L33
+export class KcSanitizer {
+    private static HREF_PATTERN = /\s+href="([^"]*)"/g;
+    private static textarea: HTMLTextAreaElement | null = null;
+
+    public static async sanitize(html: string | null): Promise<string> {
```

Comment:
The `sanitarize` method needs to be synchronous; otherwise, I won't be able to integrate it.  
Let's focus on building what we need today and not accommodate for a potential future SSR implementation.  
Lazy import won't work here, you have to do optional dependency injection.  

> "But I wanted to emulate the exact same code Keycloak uses."

I agree that maintaining strict parity in how things are done, especially in string sanitization, is highly valuable. It puts an end to any debates.

Reply:
Ok thanks a lot,  

I'll will ship it with #653